lib/haj_web/live/group_admin_live.ex

```diff
@@ -34,19 +45,37 @@ defmodule HajWeb.GroupAdminLive do
   end
 
   def handle_event("update_role", %{"role_form" => %{"role" => role}}, socket) do
-    {:noreply, assign(socket, role: role)}
+    {:noreply, assign(socket, role: role, matches: [])}
   end
 
-  def handle_event("add", _, %{assigns: %{matches: []}} = socket) do
+  def handle_event("add_user", _, %{assigns: %{matches: []}} = socket) do
     {:noreply, socket}
   end
 
-  def handle_event("add", _, %{assigns: %{matches: [user | _]}} = socket) do
-    add_user(user.id, socket.assigns.show_group.id, socket.assigns.role)
-
-    updated = Haj.Spex.get_show_group!(socket.assigns.show_group.id)
-
-    {:noreply, socket |> assign(show_group: updated, matches: [], query: nil)}
+  def handle_event("add_user", _, %{assigns: %{matches: [user | _]}} = socket) do
+    case Enum.any?(socket.assigns.show_group.group_memberships, fn member ->
```

Comment:
Because we are checking `socket.assigns.show_group.group_memberships` for memberships, we need to add the membership there when adding a new user. The alternative is, which I would much prefer, to do a database lookup here to check if they are members. A function could be `Haj.Spex.is_member_of_show_group?(user_id, show_group_id)` for example. Then we won't need to keep `group_memberships` in the socket either